graphql/resolve/mutation_rewriter.go

```diff
@@ -1383,11 +1383,34 @@ func rewriteObject(
 				// TODO: Add a function for parsing idVal. This is repeatitive
 				switch xid.Type().Name() {
 				case "Int":
-					val, _ := xidVal.(int64)
-					xidString = strconv.FormatInt(val, 10)
+					switch xidVal.(type) {
+					case json.Number:
+						val, _ := xidVal.(json.Number).Int64()
+						xidString = strconv.FormatInt(val, 10)
+					default:
+						val, _ := xidVal.(int64)
+						xidString = strconv.FormatInt(val, 10)
+					}
+				case "Int64":
+					switch xidVal.(type) {
+					case json.Number:
+						val, _ := xidVal.(json.Number).Int64()
+						xidString = strconv.FormatInt(val, 10)
+					case int64:
```

Comment:
This part is different for int64 than for int or float. Add a comment here on why that is so and the fact that default handles the case for string.